src/sortStudents.ts

```diff
@@ -1,16 +1,80 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = "Name",
+  Surname = "Surname",
+  Age = "Age",
+  Married = "Married",
+  AverageGrade = "AverageGrade",
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = "asc" | "desc";
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
+
+  let averageGradeA;
+  let averageGradeB;
+
+  sortedStudents.sort((a, b) => {
+    let aValue, bValue;
+
+    switch (sortBy) {
+      case SortType.Name:
+        aValue = a.name;
+        bValue = b.name;
+        break;
+
+      case SortType.Surname:
+        aValue = a.surname;
+        bValue = b.surname;
+        break;
+
+      case SortType.Age:
+        aValue = a.age;
+        bValue = b.age;
+        break;
+
+      case SortType.Married:
+        aValue = a.married;
+        bValue = b.married;
+        break;
```

Comment:
you can combine them
```suggestion
    switch (sortBy) {
      case SortType.Name:
      case SortType.Surname:
      case SortType.Age:
      case SortType.Married:
        aValue = a[sortBy];
        bValue = b[sortBy];
        break;
```